pkg/kv/kvserver/rangefeed/processor_test.go

```diff
@@ -268,11 +291,43 @@ func withSpan(span roachpb.RSpan) option {
 	}
 }
 
-func makeIntentScannerConstructor(rtsIter storage.SimpleMVCCIterator) IntentScannerConstructor {
-	if rtsIter == nil {
-		return nil
+type blockingScanner struct {
+	wrapped IntentScanner
+
+	block chan interface{}
+	done  chan interface{}
+}
+
+func (s *blockingScanner) ConsumeIntents(
+	ctx context.Context, startKey roachpb.Key, endKey roachpb.Key, consumer eventConsumer,
+) error {
+	if s.block != nil {
+		<-s.block
```

Comment:
nit: consider selecting on `ctx.Done()` here too, to respect context cancellation on test failures and avoid hangs.